blaze/expr/collections.py

```diff
@@ -134,9 +134,17 @@ def sort(child, key=None, ascending=True):
             if k not in child.dshape.measure.names:
                 msg = "sort key {} is not a column of schema {}"
                 raise ValueError(msg.format(k, child.dshape.measure))
+        elif isinstance(k, Field):
+            if k._resources() != child._resources():
```

Comment:
I think the better check is just `not k._child.isindentical(child)`, we can always be more relaxed later.